### DISCUSSION TITLE
Making changes to the backend and third party API intergration

### DISCUSSION THREAD

**Discussion by Muzho** [opencrvs/opencrvs-core#4089] [Technical Q&A] [answered]
Hello opencrvs team, i need some help making changes to the bad end...would you mind helping out on how to proceed...i also want to integrate a third-party API...
Your help is much appreciated

**Comment by Zangetsu101** [comment]
Hi @Muzho, 
We have documentation about interoperability here: https://documentation.opencrvs.org/technology/interoperability. I hope that helps.

**Comment by euanmillar** [comment, marked as answer]
Hi @Muzho there are really various things you can do and it really depends on what changes you want to make and why you want to interoperate.

Can you explain what changes you need and what kind of API you would like to integrate and perhaps share with us a potential [sequence diagram](https://www.websequencediagrams.com/)?  

Your options are:

1. Create a [system client ](https://documentation.opencrvs.org/technology/interoperability/authentication-and-authorization)and connect to a [webhook](https://documentation.opencrvs.org/technology/interoperability/webhooks)
2. Follow [OpenHIM](http://openhim.org/docs/introduction/about) documentation to create a mediator and host it.  I recommend you code this in the country configuration repo.
3. You can create an entirely custom server in the country configuration repo and add it to Docker Compose files and configure secrets there or in the country configuration playbook.  An example of this is this [PR](https://github.com/opencrvs/opencrvs-farajaland/pull/328) ... I have been working on an example recently for a MOSIP mediator to save details from MOSIP back into a registration.

If you can explain more about what you need then we can advise you better.  We available to arrange a call to discuss and can train you or even help develop what you need.  Please email us if you like: [team@opencrvs.org](mailto:team@opencrvs.org?subject:WebsiteEnquiry)

Hope this helps.

**Comment by euanmillar** [comment]
@Muzho I should also advise you that if you want to make changes to core, then follow a git [forking](https://documentation.opencrvs.org/setup/3.-installation/3.2-set-up-your-own-country-configuration/3.2.1-fork-your-own-country-configuration-repository) pattern that will easily comply with our [releasing branches](https://documentation.opencrvs.org/general/releases).  You want to make **master-<your country alpha 3 code>** and **develop-<your country alpha 3 code>** branches and use them for your changes.  In this way you can git pull and rebase from our core **master** and **develop** as we introduce new features.  This way, you can do your own thing and at the same time get our latest features and bugfixes.
